### DISCUSSION TITLE
Add data partition to standard and AB rootfs

### DISCUSSION THREAD

**Discussion by albertbu** [OE4T/meta-tegra#1643] [General]
HI, 
I'd like to add a data partition to the standard (external|internal) filesystem. 
I found the ROOTFSPART_SIZE_DEFAULT variable, but I am not sure about its effects.
Where should I Iook? Any pointers? 
Thanks

**Comment by minicarpet** [comment]
I am not sure why ROOTFSPART_SIZE_DEFAULT is configure as 55 GiB by default for the Orin NX (from file orin-nx.inc)
This actually flash 30GiB on my target.
I have mounted a 512 GB SSD on the devkit:
I was expecting to flash only the real rootfs size and other parts are mounted as user data (through fstab or something). Or do I really need to configure ROOTFSPART_SIZE_DEFAULT as my real SSD size (in shuch case, the flash will take really long time).

May I missunderstood something about rootfs ?

**Reply by dwalkes** [reply]
> I am not sure why ROOTFSPART_SIZE_DEFAULT is configure as 55 GiB by default for the Orin NX (from file orin-nx.inc)
> This actually flash 30GiB on my target.

If you `USE_REDUNDANT_FLASH_LAYOUT_DEFAULT` (which is the default for tegra-demo-distro recent branches) the rootfs partition size will be half of `ROOTFS_PART_SIZE_DEFAULT`.  See https://github.com/OE4T/meta-tegra/wiki/Redundant-Rootfs-A-B-Partition-Support

> I was expecting to flash only the real rootfs size and other parts are mounted as user data (through fetab or something). 

Look at the partition .xml file for whatever target you are using, this is what ultimately controls what goes where, and you'll need to edit this file to match your specific layout requirements.

> Or do I really need to configure ROOTFSPART_SIZE_DEFAULT as my real SSD size (in shuch case, the flash will take really long time).

If you use [initrd-flash](https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support) with bmap-tool as described there you can speed up the flash time for sparse files.
